llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2049,6 +2050,16 @@ Instruction *InstCombinerImpl::foldICmpOrConstant(ICmpInst &Cmp,
   }
 
   Value *OrOp0 = Or->getOperand(0), *OrOp1 = Or->getOperand(1);
+
+  // (icmp eq/ne (or disjoint x, C0), C1)
```

Comment:
I am not sure whether we should put the logic in `InstCombinerImpl::foldICmpBinOpEqualityWithConstant` or not.

Reply:
We have some other equality folds here, I think we should move them all when/if we move them.

Reply:
We can't move w.o a non-trivial refactor. Otherwise we hit the `X | C == C1` -> `X & ~C == C1 ^ C0` fold first.